sound/soc/sof/sof-audio.c

```diff
@@ -755,6 +755,10 @@ static int sof_tear_down_left_over_pipelines(struct snd_sof_dev *sdev)
 				if (ret < 0)
 					return ret;
 
+				ret = snd_sof_pcm_platform_hw_free(sdev, substream);
+				if (ret < 0)
+					return ret;
+
```

Comment:
If we have a sequence that's the same as in the previous patch, should we have a helper or call snd_sof_pcm_platform_hw_free() from sof_pcm_dsp_pcm_free

It seems like we can refactor a bit, no?

Reply:
ack, will do